extensions/ql-vscode/src/data-extensions-editor/external-api-usage-query.ts

```diff
@@ -0,0 +1,109 @@
+import { CoreCompletedQuery, QueryRunner } from "../queryRunner";
+import { qlpackOfDatabase } from "../contextual/queryResolver";
+import { file } from "tmp-promise";
+import { writeFile } from "fs-extra";
+import { dump as dumpYaml } from "js-yaml";
+import { getOnDiskWorkspaceFolders } from "../helpers";
+import { Logger, TeeLogger } from "../common";
+import { CancellationToken } from "vscode";
+import { CodeQLCliServer } from "../cli";
+import { DatabaseItem } from "../local-databases";
+import { ProgressCallback } from "../progress";
+
+export type RunQueryOptions = {
+  cliServer: Pick<CodeQLCliServer, "resolveQlpacks" | "resolveQueriesInSuite">;
```

Comment:
Does it make sense to define an interface for this and the other two fields where we use `Pick`? 

We might want to take this as tech debt instead of changing anything now (considering that we're at prototyping phase).

In general I think `Pick` can be really useful but it can hurt readability. To me it usually indicates fat types that could potentially be broken down.

Reply:
I think this could indeed be seen as tech debt. The CLI server type is really large, and I chose to use `Pick` here so that we don't need to mock the entire CLI server object. Even if we do create an interface out of this, we'd still be relying on types from the CLI server (like `QlpacksInfo`), so I'm not sure creating a separate interface for this would even be beneficial without larger changes to those types in general.